### PR TITLE
Add timeout to all other request functions

### PR DESCRIPTION
**Pull request by PythonCoderAS**

### CODE REVIEW

File: prawcore/requestor.py [hunk 1]
```diff
@@ -47,9 +47,11 @@ def close(self):
         """Call close on the underlying session."""
         return self._http.close()
 
-    def request(self, *args, timeout=TIMEOUT, **kwargs):
+    def request(self, *args, timeout=None, **kwargs):
         """Issue the HTTP request capturing any errors that may occur."""
         try:
-            return self._http.request(*args, timeout=timeout, **kwargs)
+            return self._http.request(
```
Comment (bboe) [inline prawcore/requestor.py:53]: Why is this changing?

Comment (PythonCoderAS) [reply prawcore/requestor.py:53]: The parent requestor functions have a default of None, so I decided to change this as well.

Comment (bboe) [reply prawcore/requestor.py]: The problem with doing it this way is that `None` then cannot be passed in to disable our default timeout.

**Comment by bboe** [comment]
Thanks! 🎆

**Pull request merged**